### PR TITLE
Adding cpp ctests

### PR DESCRIPTION
**Pull request by pooyan-dadvand**
This will automatically detect any cpp in the cpp_solver folder. Using the ctest botton in the Code status bar restuls in:

```
[proc] Executing command: /usr/bin/ctest -j34 -C Debug -T test --output-on-failure
[ctest] Cannot find file: /home/pooyan/Developments/CoSimIO/build/DartConfiguration.tcl
[ctest] Cannot find file: /home/pooyan/Developments/CoSimIO/build/DartConfiguration.tcl
[ctest]    Site: 
[ctest]    Build name: (empty)
[ctest] Test project /home/pooyan/Developments/CoSimIO/build
[ctest]     Start 1: hello_cpp_test
[ctest] 1/1 Test #1: hello_cpp_test ...................   Passed    0.01 sec
[ctest] 
[ctest] 100% tests passed, 0 tests failed out of 1
[ctest] 
[ctest] Total Test time (real) =   0.01 sec
[ctest] CTest finished with return code 0
```

### CODE REVIEW

**Comment by pooyan-dadvand** [comment]
We may put the macro in the Cosimio or in a checks.hpp in the tests folder

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -26,6 +26,21 @@ set_target_properties(test_runner_cpp PROPERTIES
...
+function(add_cpp_test TEST_SOURCE_FILE)
+    get_filename_component(TEST_FILENAME ${TEST_SOURCE_FILE} NAME)
+    string(REGEX REPLACE ".cpp" "_cpp_test" TEST_NAME ${TEST_FILENAME})
+    message(STATUS  "adding ${TEST_NAME} ")
+    add_executable(${TEST_NAME} ${TEST_SOURCE_FILE})
+    add_test(${TEST_NAME} ${TEST_NAME})
+    install(TARGETS ${TEST_NAME} DESTINATION "${CMAKE_SOURCE_DIR}/bin")
+endfunction()
+
+file(GLOB cpp_test_files "solver_integration/cpp_solver/*.cpp")
```
Comment (philbucher) [inline tests/CMakeLists.txt:38]: I would put this into the cmake in the `tests` folder

File: tests/solver_integration/cpp_solver/hello.cpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
+// CoSimulation includes
+#include "co_sim_io.hpp"
+
+// move this macro to a checks.hpp in tests folder?
```
Comment (philbucher) [inline tests/solver_integration/cpp_solver/hello.cpp:4]: I would leave those examples really self-contained, hence I would copy this to every example

File: tests/CMakeLists.txt (hunk 1 above)
Comment (pooyan-dadvand) [reply tests/CMakeLists.txt:38]: Isn't it in that cmake?

**Comment by pooyan-dadvand** [comment]
@philbucher is the c++11 switch enabled for macOS?

File: tests/CMakeLists.txt (hunk 1 above)
Comment (philbucher) [reply tests/CMakeLists.txt:38]: ah
it is 
sorry for the noise :D

**Comment by philbucher** [comment]
> @philbucher is the c++11 switch enabled for macOS?

in general yes, but you are adding a new executable for which the support needs to be enabled (afaik) individually
TBD

**Pull request merged**